### DISCUSSION TITLE
Blank date after prompting for date

### DISCUSSION THREAD

**Discussion by bayradberry** [chhoumann/quickadd#330] [Ideas]
If I prompt for a due date in a Capture, then on running the capture I leave the due date blank, can the plugin remove the icon and put nothing, instead of "📅 undefined"?

**Comment by chhoumann** [comment]
Coming back to this one very late - sorry about that.

This shipped in v2.13.0 as the `|optional` flag. Add it to the date token and leaving the prompt blank (or hitting Skip) resolves to nothing instead of `undefined`. To make the 📅 icon disappear along with the date, move it inside the format as a bracket literal:

```markdown
- [ ] {{VALUE}} {{VDATE:due,[📅 ]YYYY-MM-DD|optional}}
```

Answered, that renders `📅 2026-06-14`; skipped, it renders nothing at all - the emoji vanishes with the date.

Docs: https://quickadd.obsidian.guide/docs/FormatSyntax#optional-fields

Closing as resolved.
